### PR TITLE
Prevent redirecting ajax calls

### PR DESCRIPTION
**Pull request by tgv604**
This PR fixes an issue appearing when ajax-based form elements are present on the user profile form. Ajax calls made through use of com_ajax return profile editing pages instead of whatever they're meant to return, potentially breaking the form.

Please note there's no such elements in vanilla Joomla setup, but they may be added with custom user profile plugins (example: states/provinces dropdown list refreshing on country change, async loading of transactions history from 3rd party resources etc.)

Pull Request for Issue # n/a

### Summary of Changes

Added a check for com_ajax to stop redirections to profile for ajax calls

### Testing Instructions

requires custom user plugin/s with ajax-based form elements (the issue only affects such elements)

### Expected result

ajax calls returning their payload (json structures, html, etc.)

### Actual result

ajax calls returning profile editing page html

### Documentation Changes Required

n/a

### CODE REVIEW

**Comment by tgv604** [comment]
To clarify: one of my sites with privacy plugin enabled has also an address profile tab with nice dynamic country & states dropdowns, which work through com_ajax -- and they completely mess up the form when the privacy consent isn't given yet.

File: libraries/cms/html/select.php [hunk 1]
```diff
@@ -488,7 +488,7 @@ public static function option($value, $text = '', $optKey = 'value', $optText =
 
 		$obj = new stdClass;
 		$obj->{$options['option.key']}  = $value;
-		$obj->{$options['option.text']} = trim($text) ? $text : $value;
+		$obj->{$options['option.text']} = trim($text) != '' ? $text : $value;
```
Comment (Bakual) [inline libraries/cms/html/select.php]: This one looks unrelated and would affect all instances of a select. Can you revert it?

Comment (tgv604) [reply libraries/cms/html/select.php]: Yep it's unrelated, I wanted it to be a separate pr but messed up something.. just a moment.

Comment (tgv604) [reply libraries/cms/html/select.php]: Reverted today's commits

Comment (Bakual) [reply libraries/cms/html/select.php]: Looks worse than before. I think you messed up something real now.

Comment (tgv604) [reply libraries/cms/html/select.php]: OK I give up. I've no idea how to cleanly roll it all back now, followed github manual to the letter. Sorry. Can you do it from your side or maybe kill this PR altogether?

Comment (Bakual) [reply libraries/cms/html/select.php]: I can't do it from here. But you can always just create a new PR and close this one.

Comment (tgv604) [reply libraries/cms/html/select.php]: Sounds good, thanks
